onnxruntime/core/framework/utils.cc

```diff
@@ -446,31 +474,35 @@ static common::Status CopyInputsAcrossDevices(const SessionState& session_state,
 
   new_feeds.resize(num_feeds);
   std::vector<IDataTransfer::SrcDstPair> batched_data_transfers;
-  batched_data_transfers.reserve(num_feeds);
+  std::vector<IDataTransfer::SparseSrcDstPair> batched_sparse_data_transfers;
 
   for (size_t idx = 0; idx < num_feeds; ++idx) {
     ORT_RETURN_IF_ERROR(BatchOrCopyMLValue(session_state, copy_info[idx], orig_feeds[idx], new_feeds[idx],
-                                           &batched_data_transfers));
+                                           &batched_data_transfers, &batched_sparse_data_transfers));
   }
 
   if (!batched_data_transfers.empty()) {
     ORT_RETURN_IF_ERROR(session_state.GetDataTransferMgr().CopyTensors(batched_data_transfers));
   }
 
+  if (!batched_sparse_data_transfers.empty()) {
+    ORT_RETURN_IF_ERROR(session_state.GetDataTransferMgr().CopyTensors(batched_sparse_data_transfers));
+  }
+
   return Status::OK();
 }
 
 // public method to do a single copy. used by external partners
 common::Status CopyOneInputAcrossDevices(const SessionState& session_state, const std::string& input_name,
                                          const OrtValue& orig_mlvalue, OrtValue& new_mlvalue) {
-  if (!orig_mlvalue.IsTensor()) {
+  if (!orig_mlvalue.IsTensor() && !orig_mlvalue.IsSparseTensor()) {
     new_mlvalue = orig_mlvalue;
     return Status::OK();
   }
 
   MLValueCopyInfo copy_info;
   ORT_RETURN_IF_ERROR(CalculateStaticCopyInfoForFeed(session_state, input_name, copy_info));
-  copy_info.source_device = orig_mlvalue.Get<Tensor>().Location().device;
+  copy_info.source_device = (orig_mlvalue.IsTensor()) ? orig_mlvalue.Get<Tensor>().Location().device : orig_mlvalue.Get<SparseTensor>().Location().device;
 
```

Comment:
nit: line length should be 120 or less.

Reply:
This is formatted by clang formatter. I have been fighting with it.